src/extension.ts

```diff
@@ -973,23 +977,51 @@ const checkForTelemetry = (sensorState: any) => {
 };
 
 const updatePythonExtraPaths = () => {
-    const pathToLib: string = __dirname;
-    const currentExtraPaths: string[] =
-        vscode.workspace
-            .getConfiguration()
-            .get("python.autoComplete.extraPaths") || [];
-    if (!currentExtraPaths.includes(pathToLib)) {
-        currentExtraPaths.push(pathToLib);
-    }
-    vscode.workspace
-        .getConfiguration()
-        .update(
-            "python.autoComplete.extraPaths",
-            currentExtraPaths,
-            vscode.ConfigurationTarget.Global
-        );
+  updateConfigLists("python.autoComplete.extraPaths",[__dirname], vscode.ConfigurationTarget.Global)
 };
 
+const updatePylintArgs = (context: vscode.ExtensionContext) => {
+  const outPath:string = createEscapedPath([
+                          context.extensionPath,
+                          CONSTANTS.FILESYSTEM.OUTPUT_DIRECTORY
+                        ])
+  const pyLibsPath:string = createEscapedPath([
+                          context.extensionPath,
+                          CONSTANTS.FILESYSTEM.OUTPUT_DIRECTORY,
+                          CONSTANTS.FILESYSTEM.PYTHON_LIBS_DIR
+                        ])
+
+  // update pylint args to extend system path 
+  // to include python libs local to extention
+  updateConfigLists("python.linting.pylintArgs",
+                    [
+                        "--init-hook",
+                        `import sys; sys.path.extend([\"${outPath}\",\"${pyLibsPath}\"])`
+                    ], 
+                    vscode.ConfigurationTarget.Workspace)
+}
+
+const createEscapedPath = (pieces: string[]) => {
+  const initialPath:string = pieces.join("\\");
+
+  // escape all instances of backslashes
+  return initialPath.replace(/\\/g,"\\\\");
```

Comment:
You can also use:
 
```python
import re
re.escape(str)
```

Assuming we want to escape for regex?

Reply:
I tried to use escape(), but it also escaped other characters (like semicolons) :(

Reply:
^ oh wait, that's Python (I think). I'm using Typescript.

Reply:
We talked offline, we don't really need to escape this for regex, but rather just the backlashes.